services-core/src/economic_viability.rs

```diff
@@ -298,4 +363,61 @@ mod tests {
 
         assert_eq!(priority_min_avg_fee.min_average_fee().wait().unwrap(), 42);
     }
+
+    #[test]
+    fn bounded_uses_min_of_first_and_second_min_avg_fee() {
+        let mut first = MockEconomicViabilityComputing::new();
+        first.expect_min_average_fee().return_once(|| Ok(10));
+        let mut second = MockEconomicViabilityComputing::new();
+        second.expect_min_average_fee().return_once(|| Ok(5));
+        let bounded = BoundedEconomicViabilityComputer::new(Box::new(first), Box::new(second));
+        assert_eq!(
+            bounded.min_average_fee().now_or_never().unwrap().unwrap(),
+            5
+        );
+    }
+
+    #[test]
+    fn bounded_uses_max_of_first_and_second_max_gas_price() {
+        let mut first = MockEconomicViabilityComputing::new();
+        first.expect_max_gas_price().return_once(|_| Ok(10.into()));
+        let mut second = MockEconomicViabilityComputing::new();
+        second.expect_max_gas_price().return_once(|_| Ok(5.into()));
+        let bounded = BoundedEconomicViabilityComputer::new(Box::new(first), Box::new(second));
+        assert_eq!(
+            bounded
+                .max_gas_price(Default::default())
+                .now_or_never()
+                .unwrap()
+                .unwrap(),
+            10.into(),
+        );
+    }
+
+    #[test]
+    fn bounded_fails_if_second_fails() {
+        let mut first = MockEconomicViabilityComputing::new();
+        first.expect_min_average_fee().return_once(|| Ok(10));
+        let mut second = MockEconomicViabilityComputing::new();
+        second
+            .expect_min_average_fee()
+            .return_once(|| Err(anyhow!("")));
+        let bounded = BoundedEconomicViabilityComputer::new(Box::new(first), Box::new(second));
+        assert!(bounded.min_average_fee().now_or_never().unwrap().is_err());
+    }
+
+    #[test]
+    fn bounded_uses_second_on_error() {
+        let mut first = MockEconomicViabilityComputing::new();
+        first
+            .expect_min_average_fee()
+            .return_once(|| Err(anyhow!("")));
+        let mut second = MockEconomicViabilityComputing::new();
+        second.expect_min_average_fee().return_once(|| Ok(5));
+        let bounded = BoundedEconomicViabilityComputer::new(Box::new(first), Box::new(second));
+        assert_eq!(
+            bounded.min_average_fee().now_or_never().unwrap().unwrap(),
+            5
+        );
+    }
```

Comment:
This file is getting quite large. Maybe it makes sense to split up the three implementations into separate submodules?